src/psyclone/tests/psyir/frontend/fparser2_test.py

```diff
@@ -614,6 +614,46 @@ def test_process_declarations():
     assert isinstance(ptr_sym.initial_value, IntrinsicCall)
 
 
+@pytest.mark.usefixtures("f2008_parser")
+def test_process_declarations_class_keyword():
+    '''Test that process_declarations method of Fparser2Reader
+    converts the fparser2 declarations with the class keyword to symbols
+    in the provided parent Kernel Schedule.
+
+    '''
+    fake_parent = KernelSchedule.create("dummy_schedule")
+    processor = Fparser2Reader()
+
+    # Class declaration
+    reader = FortranStringReader("class(my_type), intent(in) :: carg")
+    # Set reader to free format (otherwise this is a comment in fixed format)
+    reader.set_format(FortranFormat(True, True))
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    sym = fake_parent.symbol_table.lookup("carg")
+    assert isinstance(sym, DataSymbol)
+    assert isinstance(sym.datatype, DataTypeSymbol)
+    assert sym.datatype.is_class is True
+    assert sym.name == "carg"
+    assert sym.datatype.name == "my_type"
+    assert isinstance(sym.interface, ArgumentInterface)
+    assert sym.interface.access == ArgumentInterface.Access.READ
+
+    # class(*) case
+    reader = FortranStringReader("class(*), intent(in) :: urgh")
+    reader.set_format(FortranFormat(True, True))
+    fparser2spec = Specification_Part(reader).content[0]
+    processor.process_declarations(fake_parent, [fparser2spec], [])
+    sym = fake_parent.symbol_table.lookup("urgh")
+    assert isinstance(sym, DataSymbol)
+    assert isinstance(sym.datatype, DataTypeSymbol)
+    assert sym.datatype.is_class is True
+    assert sym.name == "urgh"
+    assert sym.datatype.name == "*"
```

Comment:
As discussed in the implementation, I think this should either be UnsupportedFortranType (the easiest thing to do) or an UnresolvedFortranType with an `is_class==True` attribute.